### PR TITLE
Adding functionality for creating stats.csv

### PR DESCRIPTION
**Pull request by john-baute**
Added functionality to create stats.csv file. calculates stats for each event code. edited the get_state_dict, comp, and main functions.

### CODE REVIEW

File: CDC-Data-Reconciliation-Backend/compare.py [hunk 1]
```diff
@@ -66,7 +86,7 @@ def get_cdc_dict(cdc_file):
 
     return cdc_dict
 
-
+# place the stats stuff here
 def comp(state_dict, cdc_dict):
     for state_case_id in state_dict:
         state_row = state_dict[state_case_id]
```
Comment (Carson21) [inline CDC-Data-Reconciliation-Backend/compare.py:92]: After line 92 we should check whether the 'CaseClassStatus' is 'Confirmed' or 'Suspect', if it isn't we should skip this entry for comparison / stats. We should talk to Marion on whether 'Suspect' cases are to be reported to the CDC as well.

I believe you should create a new entry to stats if the event code is not yet in the stats dictionary. If it already is in the dictionary, add to 'totalCases'.

File: CDC-Data-Reconciliation-Backend/compare.py [hunk 2]
```diff
@@ -106,6 +133,15 @@ def comp(state_dict, cdc_dict):
...
                        cdc_row['MMWRYear'], cdc_row['MMWRWeek'], "Case ID not found in State CSV File", "4"))
+        
+        # adding in missing from state count, total case count, and caseID to the stats dict
+        # only counting cases that are not duplicates, otherwise counting as duplicate
+        if cdc_row['CaseID'] not in stats[cdc_row['EventCode']]['caseIDs']:
+            stats[cdc_row['EventCode']]['totalMissingState'] += 1
+            stats[cdc_row['EventCode']]['totalCases'] += 1
+            stats[cdc_row['EventCode']]['caseIDs'].append(cdc_row['CaseID'])
+        else:
+            stats[cdc_row['EventCode']]['totalDuplicates'] += 1
```
Comment (Carson21) [inline CDC-Data-Reconciliation-Backend/compare.py:144]: First, you are not guaranteed that the EventCode already exists within the stats dictionary, and thus you should add an entry if it is not. Also, you are guaranteed here that the CaseID will not be present within the state as this for loop is going over all caseID's that were not in the state data.

'TotalDuplicates' should be calculated in the get_cdc_dict function as we disregard duplicates from CDC there and also report if there is a CDC duplicate.

File: CDC-Data-Reconciliation-Backend/compare.py [hunk 3]
```diff
@@ -132,7 +168,24 @@ def main():
...
             writer.writerow({'CaseID': result.caseID, 'EventCode': result.eventCode, 'MMWRYear': result.MMWRYear,
                             'MMWRWeek': result.MMWRWeek, 'Reason': result.reason, 'ReasonID': result.reasonID})
-
+            
+    # writing to stats.csv but first grabbing the folder location of results.csv
+    output_directory = os.path.dirname(args.output)
+    if output_directory == '':
+        output_directory = '.'
+    
+    # i was not sure how to handle reportID in the csv, so i have left it out from this. If it needs to be added into the csv then do it here
```
Comment (Carson21) [inline CDC-Data-Reconciliation-Backend/compare.py:177]: you are correct, reportID should not be included here.

**Review by Carson21 — COMMENTED** [review]
Mainly, 'totalDuplicates' should be calculated within the get_cdc_dict function, not the get_state_dict function. Also cases that are not 'Suspect' or 'Confirmed' should be skipped in the comparison / stats.

We will need to check with Marion on whether 'Suspect' cases are to be reported to the CDC.

**Review by waffy1901 — COMMENTED** [review]
I believe Carson hit on the few things that need to be changed. I don't have any additional changes. Overall, I think this looks quite good - great work!

**Comment by Carson21** [comment]
After receiving confirmation from Marion, all cases in the state DB even if they are marked 'Suspect' or 'Not a Case / Deleted' should be included within the comparison / report.

File: CDC-Data-Reconciliation-Backend/compare.py [hunk 4]
```diff
@@ -106,6 +132,16 @@ def comp(state_dict, cdc_dict):
...
+        
+        # adding in missing from state count, total case count, and caseID to the stats dict
+        # only counting cases that are not duplicates, otherwise counting as duplicate
+        if cdc_row['EventCode'] in stats:
+            stats[cdc_row['EventCode']]['totalMissingState'] += 1
+            stats[cdc_row['EventCode']]['totalCases'] += 1
+            stats[cdc_row['EventCode']]['caseIDs'].append(cdc_row['CaseID'])
+        
+        else:
+            stats[cdc_row['EventCode']] = {'totalCases': 1, 'totalDuplicates': 0, 'totalMissingCDC': 0, 'totalMissingState': 0, 'totalWrongAttributes': 0, 'caseIDs': [cdc_row['CaseID']]}
```
Comment (Carson21) [inline CDC-Data-Reconciliation-Backend/compare.py:144]: Shouldn't totalMissingState be 1 here on line 144?

**Review by Carson21 — APPROVED** [review]
Everything looks good

**Pull request merged**